### ISSUE TITLE
Consider changing AOTAutograd cache to hit on graphs with different input and node names

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#157792]
We ran into this in vLLM, where vLLM:
1) creates one big graph with a lot of transformer layers
2) splits the big graph into multiple graphs, each of which are the same
3) compiles each graph separately (sending it through AOTAutograd). All of these graphs cache miss due to having different input names.

The workaround I am currently applying is to normalize the inputs of the graphs before sending them through AOTAutograd. It's not clear to me if the right long-term solution is:
1) AOTAutograd cache becomes agnostic to names
2) user is supposed to do some sort of hierarchical compilation
3) user is supposed to know these quirks with AOTAutograd cache and program with those in mind

**Comment by bdhirsh** [comment]
I remember @bobrenjc93 @laithsakka running into a similar problem with dynamic shape and caching (SymInts were getting different names in the graph from run to run, causing unnecessary cache misses). IIRC, they also added a pass to normalize SymInt names for that case.

**Comment by zou3519** [comment]
Repro:
- patch in https://github.com/pytorch/pytorch/pull/157803/files#diff-03de9d44b47c56dedd2bca3c1b0b21a08c93df662579165feafd49c5226fd83dR1857
- run that test
- note that AOTAutograd cache misses 3 times (it should cache hit on the last two)

**Comment by laithsakka** [comment]
For dynamic shapes what we end up doing is that when we pick the symbol names we pick them such that same code locations always get the same name.

**Comment by laithsakka** [comment]
> Repro:
> 
> * patch in https://github.com/pytorch/pytorch/pull/157803/files#diff-03de9d44b47c56dedd2bca3c1b0b21a08c93df662579165feafd49c5226fd83dR1857
> * run that test
> * note that AOTAutograd cache misses 3 times (it should cache hit on the last two)

here is a TL-parse for the repo [internal link]
https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/lsakka/bb96d84f-144b-45a7-ad53-db3f62f0f0dd/custom/index.html?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=10000

graph 1
```
class GraphModule(torch.nn.Module):
    def forward(self, l_a0_: "f32[][]cpu", l_x_: "f32[][]cpu", l_a1_: "f32[][]cpu"):
         # File: /home/lsakka/pytorch/test/inductor/test_codecache.py:1860 in forward, code: x = x + (a0**2) + (a1 / 2)
        pow_1: "f32[][]cpu" = l_a0_ ** 2;  l_a0_ = None
        add: "f32[][]cpu" = l_x_ + pow_1;  l_x_ = pow_1 = None
        truediv: "f32[][]cpu" = l_a1_ / 2;  l_a1_ = None
        add_1: "f32[][]cpu" = add + truediv;  add = truediv = None
        return [add_1]
        
```

graph 2
```
class GraphModule(torch.nn.Module):
    def forward(self, l_b0_: "f32[][]cpu", x: "f32[][]cpu", l_b1_: "f32[][]cpu"):
         # File: /home/lsakka/pytorch/test/inductor/test_codecache.py:1861 in forward, code: x = x + (b0**2) + (b1 / 2)
        pow_1: "f32[][]cpu" = l_b0_ ** 2;  l_b0_ = None
        add: "f32[][]cpu" = x + pow_1;  x = pow_1 = None
        truediv: "f32[][]cpu" = l_b1_ / 2;  l_b1_ = None
        add_1: "f32[][]cpu" = add + truediv;  add = truediv = None
        return [add_1]
        

 # graph id: 139846230953600
```

**Comment by jamesjwu** [comment]
Just to be clear, in this case the names are different but there are no symints involved; the actual sources for these tensors are different and differently named. It does seem like the nodes of the graph are always the same name, just the inputs aren't normalized to the same name. Will think about how to fix this on aotautogradcache side.

**Comment by zou3519** [comment]
In the vLLM case there are also symints in the graph, I didn't put together a repro with that
